### ISSUE TITLE
Changing the Time scale

### ISSUE THREAD

**Issue by ScottW1** [apollo-ng/picoReflow#5]
Here is the question I posted on your other site (as requested).  I assumed that since this wasn't an error in the code, it wasn't a true "issue".  I posted my original post as well as your reply and my response:

scott
2014/11/23 19:46

This is great! I've been pouring over the code trying to convert it to a ceramic kiln controller. To do so, I need to change the time scale from seconds to minutes. I would assume that I would be able to add a *60 scale to where the code reads the profiles, but cant seem to see where to input it. Any tips? I'm fine with the code running in seconds (even if its a bit overkill for the thermal inertia my kiln has), but I would like the display and profile input to be in minutes. A 12 hour run on the kiln becomes a bit silly in seconds :)

Thanks for your help and awesome code!

---

chrono
2014/11/23 20:00

Do you mean in the python or in the UI part of the code? We've opted for the most efficient integer SI unit for storage and calculation, which are seconds. The UI part was originally designed for Reflow Curves, which are short enough to be displayed in seconds for curve/datasheet comparison. However, in your case, I can see that keeping the legends and UI stuff in seconds would be more than a bit silly, when you're going over several hours.

If you have a github account, I'd love if you would open a new issue, so we and possibly others can keep track of this issue (history and solutions). If not, don't worry about it, I would leave the data in seconds as it is, I think the solution rather lies in the datapoint/legend aggregation of the plotting framework we used and maybe a couple of *60 in the javascript for the curve detail panel. I'll have to have a look into the code and will come back to you, either here via comment (don't forget to subscribe) or via github, if we have a new issue :)

---

scott
2014/11/24 8:27

Yes, my thought was to leave the Python in seconds (unless its easier not to), and modify the Java so that the interface (display and profile entry) are in hours and minutes.  I am fairly mediocre in Java at the moment, so any help would be great!  I can see how to change some of the display points (like total duration), but I have a problem translating the input from the profile file.Thanks for your help!

**Comment by chron0** [comment]
Nice, I'm happy that it kinda works for you :) Now we just have to iron out the bugs. Using PID for temperature control is actually completely overblown, as a german idiom would say, firing cannons at sparrows but it's a proven and reliable technology, so we figured it wouldn't hurt.

**Comment by ScottW1** [comment]
Is there an easy way in the code, like add a time multiplier, to only store data points every 5 seconds or so?  Even 30 seconds would probably be fine in my situation. That might alleviate the resource drain of having millions of data points after a long run.

**Comment by chron0** [comment]
Nope, not yet anyways and I don't think that is really the issue. Of course we could think about building some input throttling but today storage is cheap and available and leveldb is is pretty robust and scales very well up and down. The issue is the aggregation when querying, which has to kick in automatically if the query would return a dataset that is too big for the browser to handle so that we can have both, a macroscopic view, showing the whole timeframe aggregated to viewer points and the ability to "zoom" in, with full data resolution in a microscopic view which might help with other things.
